### ISSUE TITLE
DMR direction from DMLtest.multifactor

### ISSUE THREAD

**Issue by jdilme** [haowulab/DSS#26]
Hello and thank you for this beautiful package.

TL;DR: I would like to know if it is possible to determine DMR direction using the DMLtest object from the multifactor function. I understand it doesn't make sense if multiple factors were used, but I am testing a contrast with two factors.

For context, it is RRBS data and I have the samples divided into three groups (call it factor A) and within each group one more factor (call it B) with 4 levels.
I am interested in the pairwise comparison of factor B levels within each group A.
I decided to use the multifactor approach as I thought it was more correct to take into account all the coverage variability of the samples within each group A.

I also performed smoothing when fitting the model as I have no biological replicates for most of the levels of B.

**Comment by haowulab** [comment]
The result from callDMR function has a column `diff.Methy` for two-group comparison. Its sign represents the direction of the DMRs.

**Comment by jdilme** [comment]
I believe it is not the case if the DMLobject is obtained from the
DMLtest.multifactor() fumction. Maybe it is just not possible to estimate
the methylation levels using the multifactor approach.

**Comment by haowulab** [comment]
Sorry I misread your question. I thought you were doing two-group comparison using DMLtest. I now understand that you put all 4 groups together in a multiple regression model, but do pairwise comparison from that. In that case, the callDMR function returns a data frame with `areaStat`. Its sign indicates the direction.

**Comment by jdilme** [comment]
Thank you for the quick responses!

Yes exactly, that's what I did
 I already looked at the areaStat output to see if it had a direction
indication with sign as when used with DMLtest, but when using the
multifactor design it only outputs positive values.

**Comment by jdilme** [comment]
Thank you for your quick responses, exactly, that is what I did. That's why I was asking, because I am not interested in the methylation levels, only the direction of the DMRs, and the areaStat doesn't seem to give me the direction either.

Here I attatch the output of a Summary of this column, as you can see, there are no negative values.
```
summary(dmrs$areaStat)
    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
   9.637   29.875   68.197  112.424  145.538 1021.938 
```

**Comment by haowulab** [comment]
I'm not sure if you are using the latest version. If you run the last few lines of the examples in the function help for `callDMR`: 

```
     data(RRBS)
     DMLfit = DMLfit.multiFactor(RRBS, design, ~case+cell+case:cell)
     DMLtest.cell = DMLtest.multiFactor(DMLfit, coef="cellrN")
     dmr = callDMR(DMLtest.cell, p.threshold=0.05) 
     dmr
```

Below is what I get, where `areaStat` has signs:

```
> dmr
     chr   start     end length nCG   areaStat
33  chr1 2793724 2793907    184   5  12.619968
413 chr1 3309867 3310133    267   7 -12.093850
250 chr1 3094266 3094486    221   4  11.691413
262 chr1 3110129 3110394    266   5  11.682579
180 chr1 2999977 3000206    230   4  11.508302
121 chr1 2919111 2919273    163   4   9.421873
298 chr1 3146978 3147236    259   5   8.003469
248 chr1 3090627 3091585    959   5  -7.963547
346 chr1 3200758 3201006    249   4  -4.451691
213 chr1 3042371 3042459     89   5   4.115296
169 chr1 2995532 2996558   1027   4  -2.988665
```

**Comment by jdilme** [comment]
That's interesting, I am using the development version (2.45). And tested with your data and got the same results as you, so the version is not the problem.

It may be that I am using a contrast matrix instead of coef? I can't use coef for my experiment, so that may be the problem.

I will try to estimate the methylation direction by intersecting the DMLs with the DMRs and extract the methylation data from my original RRBS object.

If there is anything I can modify to obtain the direction please let me know and thank you for your help :)

**Comment by haowulab** [comment]
Oh yes. If you specify a contrast matrix, it'll become an F-test. You know F-test statistics are always positive, so you will lose the sign. The same goes for specifying `term`. When testing a whole term in the formula, it'll do an F-test, because a categorical term can contain multiple levels. 

There must be a way to test `coef` for your purposes. Let me try to figure out and get back to you.

**Comment by haowulab** [comment]
Okay, I found that I actually tried to attach a sign to the areaStat even from F-test. But there's a silly bug in the code so it doesn't work. I just corrected it and it provides signs now. If you give me an email I'll send you the new package. I just committed the change to bioconductor, but it could take a couple days or so for the new version to appear on devel.

**Comment by jdilme** [comment]
Oh really? That would be awesome, thank you so much. Could you send it to
***@***.*** ?


Thank you once again for your help, it is an amazing tool :). Have a nice
day.

**Comment by haowulab** [comment]
Email seems to be blocked by GitHub. All I see is "***@***.*** ". Please send me an email, and I'll email back. You can find my email on my website haowulab.org.

**Comment by jdilme** [comment]
Have you received my email? Thank you:)

**Comment by haowulab** [comment]
I just received your email and I sent the package again. Let me know if you received it.

**Comment by haowulab** [comment]
It's weird that you can't receive my email. Let's try this. I put the package on dropbox. See if you can download from https://www.dropbox.com/s/u1kmv4ud6rs8vqb/DSS_2.44.0.tar.gz?dl=0.

**Comment by jdilme** [comment]
It works like a charm! Thank you for your help :)
